sdk/core/System.ServiceModel.Rest/src/Result.cs

```diff
@@ -22,31 +22,44 @@ public abstract class Result
     /// <exception cref="NotImplementedException"></exception>
     public static Result<T> FromValue<T>(T value, Result result)
     {
-        return new Result<T>(value, result);
+        return new ValueResult<T>(result, value);
     }
 
+    /// <summary>
+    /// Indicates whether the status code of the returned response is considered
+    /// an error code.
+    /// </summary>
+    // TODO: we have to make IsError publicly settable, but we might not want this.  Rethink?
```

Comment:
It would be good to keep the IsError setter internal to the pipeline if we can.  If the pipeline largely moves into the System dll, we can maybe put this back.